src/mui/input/ReferenceManyInput.js

```
@@ -0,0 +1,222 @@
+ * for the query - by default { q: [searchText] }. You can customize the mapping
+ * searchText => searchQuery by setting a custom `filterToQuery` function prop:
+ *
+ * @example
+ * <ReferenceManyInput
+ *      source="post_id"
+ *      reference="posts"
+ *      filterToQuery={searchText => ({ title: searchText })}>
+ *     <SelectManyInput type="object" optionText="title" />
+ * </ReferenceManyInput>
+ */
+export class ReferenceManyInput extends Component {
+    constructor(props) {
+        super(props);
+        const {perPage, sort, filter} = props;
+        // stored as a property rather than state because we don't want redraw of async updates
+        this.params = {pagination: {page: 1, perPage}, sort, filter};
+        this.debouncedSetFilter = debounce(this.setFilter.bind(this), 500);
+    }
+
+    componentDidMount() {
+        this.fetchReferenceAndOptions();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.record.id !== nextProps.record.id) {
+            this.fetchReferenceAndOptions(nextProps);
+        }
+    }
+
+    setFilter = (filter) => {
+        if (filter !== this.params.filter) {
+            this.params.filter = this.props.filterToQuery(filter);
+            this.fetchReferenceAndOptions();
+        }
+    }
+
+    setPagination = (pagination) => {
+        if (pagination !== this.param.pagination) {
+            this.param.pagination = pagination;
+            this.fetchReferenceAndOptions();
+        }
+    }
+
+    setSort = (sort) => {
+        if (sort !== this.params.sort) {
+            this.params.sort = sort;
+            this.fetchReferenceAndOptions();
+        }
+    }
+
+    fetchReferenceAndOptions({input, reference, source, resource} = this.props) {
+        const {pagination, sort, filter} = this.params;
+        const ids = input.value;
+        if (ids && Array.isArray(ids)) {
+            this.props.crudGetMany(reference, ids);
```

you should probably throw an explicit error if `ids` is not an array.